packages/@glimmer/runtime/lib/bounds.ts

```diff
@@ -75,7 +74,7 @@ export function clear(bounds: Bounds): Nullable<SimpleNode> {
   while (true) {
     let next = current.nextSibling;
 
-    parent.removeChild(current);
+    current.parentNode?.removeChild(current);
```

Comment:
..and then in here, guard with `if (parent === current.parentNode)`. In the else branch, we may want to add two different development mode warnings for the case where it is null and for the case it is not null but different (ie the node has been moved)